### ISSUE TITLE
dill fails (_locate_function) to pickle a function created with eval()

### ISSUE THREAD

**Issue by pneykov** [uqfoundation/dill#77]
a fix is simple in the failing function `_locate_function` add a check if the `obj.__module__ is None`:
```
def _locate_function(obj, session=False):
    if not obj.__module__ or obj.__module__ == '__main__': # and session:
        return False
    found = _import_module(obj.__module__ + '.' + obj.__name__, safe=True)
    return found is obj
```

**Comment by mmckerns** [comment]
While this is a good addition -- I'm not seeing a case where `obj.__module__ is None`.  Do you have an example?  When I build `obj = eval('lambda x:x')`, I get `obj.__module__` is `'__main__'`.

**Comment by pneykov** [comment]
This reproduces the problem (I guess it is because we pass the globals parameter from the outside):

 

 

import math

 

def get_lambda(str, **kwarg):

    return eval(str, kwarg, None)

    

if __name__ == '__main__':

    obj = get_lambda('lambda x: math.exp(x)', math=math)

    print(obj.__module__)

**Comment by mmckerns** [comment]
Implemented in: 92f2db0c8220050cc5582bc4d127455096a7fff8
